packages/odyssey-react/src/utils/omit.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { useMemo } from 'react';
+
+type omit = <T, U extends T>(props: T, other?: string[]) => U;
```

Comment:
is there a bounded polymorphic limit we can apply for `T` here that react provides? [I didn't see anything sensible here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts)

Reply:
Based on what I've seen thus far most of the prop type definitions are generics that have an expectation of an element property being defined at the callsite, to keep the call sites terse for now I've reworked this to extend from an upper limit of an object as that is how TS types rest parameters